File: backend/src/rag/schemas/load_gate.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class LoadGateSettings(BaseModel):
    """Seuils du gate de charge — persistés dans admin.env, relus à chaud."""

    model_config = ConfigDict(extra="forbid")

    enabled: bool
    cpu_threshold_pct: int = Field(..., ge=1, le=100)
    memory_threshold_pct: int = Field(..., ge=1, le=100)


class LoadGateStatusOut(BaseModel):
    """Statut instantané : métriques mesurées + seuils + verdict."""

    enabled: bool
    overloaded: bool
    cpu_psi_avg60: float | None
    memory_used_pct: float | None
    cpu_threshold_pct: int
    memory_threshold_pct: int
    reasons: list[str]

File: frontend/src/lib/load-gate.ts
import { api } from "@/lib/api";

// Gate de charge serveur (enabler 01f8992b) — miroir de rag/schemas/load_gate.py.
export type LoadGateStatus = {
  enabled: boolean;
  overloaded: boolean;
  cpu_psi_avg60: number | null;
  memory_used_pct: number | null;
  cpu_threshold_pct: number;
  memory_threshold_pct: number;
  reasons: string[];
};

export type LoadGateSettings = {
  enabled: boolean;
  cpu_threshold_pct: number;
  memory_threshold_pct: number;
};

export const loadGateApi = {
  get: () => api.get<LoadGateStatus>("/api/admin/load-gate"),
  set: (settings: LoadGateSettings) => api.put<LoadGateStatus>("/api/admin/load-gate", settings),
};
